Sources/Network/HTTPS/HttpProtocolConnectionFactory.swift

```diff
@@ -19,7 +19,7 @@ class HttpProtocolConnectionFactory: ProtocolConnectionFactory {
     ) -> ConsensusHttpConnection {
         ConsensusHttpConnection(
                         config: config,
-                        requester: RestApiRequester(requester: requester, baseUrl: config.url),
+                        requester: RestApiRequester(requester: requester, baseUrl: config.urlLoadBalancer.nextUrl()),
```

Comment:
i think this is a problem because you rotate to "nextUrl" to set baseUrl, but then the config gets passed down further the chain, and will have nextUrl called again in the Inner.init(...), possibly leading to a mismatch